addons/mail/models/template_reset_mixin.py

```diff
@@ -0,0 +1,142 @@
+import os
+
+from lxml import etree
+
+from odoo import api, fields, models, tools, _
+from odoo.modules import get_module_resource
+from odoo.modules.module import get_resource_from_path, get_resource_path
+from odoo.tools.convert import xml_import
+from odoo.tools.misc import file_open
+from odoo.tools.translate import TranslationFileReader
+
+
+class TemplateResetMixin(models.AbstractModel):
+    _name = "template.reset.mixin"
+    _description = 'Template Reset Mixin'
+
+    template_fs = fields.Char(
+        string='Template Filename', copy=False,
+        help="""File from where the template originates. Used to reset broken template.""")
+    is_template_modified = fields.Boolean(string='Is Template Modified', copy=False)
+
+    # -------------------------------------------------------------------------
+    # OVERRIDE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if 'template_fs' not in vals and 'install_filename' in self.env.context:
+                # we store the relative path to the resource instead of the absolute path, if found
+                # (it will be missing e.g. when importing data-only modules using base_import_module)
+                path_info = get_resource_from_path(self.env.context['install_filename'])
+                if path_info:
+                    vals['template_fs'] = '/'.join(path_info[0:2])
+        return super().create(vals_list)
+
+    def write(self, vals):
+        vals['is_template_modified'] = vals.get('is_template_modified', True)
+        return super().write(vals)
+
+    def _load_records_write(self, values):
+        # OVERRIDE to make the fields blank that are not present in xml record
+        if self.env.context.get('reset_template'):
+            # We don't want to change anything for magic columns, values present in XML record, and
+            # special fields self.CONCURRENCY_CHECK_FIELD (__last_update) and 'template_fs'
+            fields_in_xml_record = set(values.keys())
+            fields_not_to_touch = set(models.MAGIC_COLUMNS) | fields_in_xml_record | {self.CONCURRENCY_CHECK_FIELD, 'template_fs'}
+            fields_to_empty = set(self._fields.keys()) - fields_not_to_touch
+            # For the fields not defined in xml record, if they have default values, we should not
+            # enforce empty values for them and the default values should be kept
+            field_defaults = self.default_get(list(fields_to_empty))
+            # Update the values to be written and include the default values, prevent fields with
+            # default values from being empty
+            values.update(field_defaults)
+            fields_to_empty = fields_to_empty - set(field_defaults.keys())
+            # Finally, update the values with fields that should be empty
+            values.update({field: False for field in fields_to_empty})
```

Comment:
```suggestion
            values.update(dict.fromkeys(fields_to_empty. False))
```
Or pass a gencomp to `update` so it doesn't have to reify a full dict (I assume).